feat(satp-hermes): implement API1 endpoints

AndreAugusto11 opened the pull request:
Implemented the following endpoints, with some initial tests:
* healthcheck: responds if SATP Hermes is on
* get status: retrieve the status of a SATP session
* get integrations: retrieves data about each supported network, or other external system

petermetz commented on packages/cactus-plugin-satp-hermes/src/main/typescript/blo/admin/get-healthcheck-handler-service.ts:
```diff
@@ -0,0 +1,41 @@
+import { Logger } from "@hyperledger/cactus-common";
+import { SATPManager } from "../../gol/satp-manager";
+
+export async function ExecuteGetHealthCheck(
```
@AndreAugusto11 Please use `camelCase` for function names. Applies to all functions throughout the PR/codebase

petermetz commented on packages/cactus-plugin-satp-hermes/src/main/typescript/blo/admin/get-healthcheck-handler-service.ts:
```diff
@@ -0,0 +1,41 @@
+      throw error;
+    } else {
+      logger.error(`${fnTag}, Unexpected error: ${error.message}`);
+      throw new Error("An unexpected error occurred while obtaining status.");
```
@AndreAugusto11 Very very new thing but on the main branch (as of 4 days ago) now you can use these constructs for chaining errors with their causes:
https://github.com/hyperledger-cacti/cacti/pull/3593

I'm not sure if the current base branch you are sending this PR against is up to date with main that recently, but just for at least future reference wanted to make note that you could soon (or maybe already) use the `cause` property of the `Error` built-in type as shown in the PR description of the PR linked above.
Just raising awareness.

RafaelAPB replied:
This is a neat feature

AndreAugusto11 replied:
@RafaelAPB could you please confirm that this branch is not rebased with main? In that case, I woud defer this change to a future PR when merging/rebasing with main

petermetz commented on packages/cactus-plugin-satp-hermes/src/main/typescript/blo/admin/get-healthcheck-handler-service.ts:
```diff
@@ -0,0 +1,41 @@
+import { SATPManager } from "../../gol/satp-manager";
+
+export async function ExecuteGetHealthCheck(
+  logger: Logger,
```
@AndreAugusto11 I recommend passing in the log level instead and creating your own logger with a label that matches the function name so that in the logs it's easier to find later on.

An example of how I usually do it can be seen here:
`packages/cactus-plugin-ledger-connector-besu/src/main/typescript/impl/get-block-v1/get-block-v1-impl.ts`

https://github.com/hyperledger-cacti/cacti/blob/ff842d2a59ef898e734e1424ee6f26b52ba0af9b/packages/cactus-plugin-ledger-connector-besu/src/main/typescript/impl/get-block-v1/get-block-v1-impl.ts

Applies throughout the PR/codebase as a suggestion.

petermetz reviewed (commented):
Looking good in general but I left a few suggestions for generic code quality improvement

RafaelAPB reviewed (changes requested):
LGTM apart from a few minor issues. Please make sure to incorporate  @petermetz 's feedback as well

brunoffmateus commented on packages/cactus-plugin-satp-hermes/src/main/typescript/web-services/healthcheck-endpoint.ts:
```diff
@@ -0,0 +1,96 @@
+import {
```
my file was a bit hardcoded, so this is an improvement! it seems good to me, but be sure to doublecheck @AndreAugusto11

RafaelAPB commented:
@AndreAugusto11 can you please solve conflicts?

AndreAugusto11 commented:
@RafaelAPB @petermetz thanks a lot for the comments and suggestions. I have applied them and updated the PR. The main changes were related to passing the log Level instead of the Logger itself and creating an Enum for the gateway status. Also, updated the open API specification to be consistent with the new Enum

RafaelAPB commented:
Thank you very much, great work
